src/harbour-advanced-camera.cpp

```diff
@@ -54,10 +55,20 @@ int main(int argc, char *argv[])
 
     view->rootContext()->setContextProperty("modelResolution", &resolutionModel);
     view->rootContext()->setContextProperty("sortedModelResolution", &sortedResolutionModel);
+    StorageModel storageModel;
+    view->rootContext()->setContextProperty("modelStorage", &storageModel);
+    FSOperations fsOperations;
+    view->rootContext()->setContextProperty("fsOperations", &fsOperations);
+
     view->setSource(SailfishApp::pathTo("qml/harbour-advanced-camera.qml"));
 
     QObject::connect(view, &QQuickView::focusObjectChanged, &handler,
                      &ResourceHandler::handleFocusChange);
+    QObject::connect(&fsOperations, &FSOperations::rescan, &storageModel,
+                     &StorageModel::scan);
+    QObject *btnStorage = view->rootObject()->findChild<QObject *>("btnStorage");
+    QObject::connect(btnStorage, SIGNAL(refreshStorage(QString)), &storageModel,
```

Comment:
Can this not be done with a QML connection?

Reply:
Not sure one can connect QML signal to C++ slot using Connections, but in fact didn't even need signal since I could use modelStorage directly...